src/main/java/io/reactivex/processors/MulticastProcessor.java

```diff
@@ -0,0 +1,637 @@
+    assertTrue(mp2.offer(4));
+
+    assertFalse(mp2.offer(5));
+
+    mp2.onComplete();
+
+    mp2.test().assertResult(1, 2, 3, 4);
+ * </code></pre>
+ * @param <T> the input and output value type
+ * @since 2.1.14 - experimental
+ */
+@Experimental
+@BackpressureSupport(BackpressureKind.FULL)
+@SchedulerSupport(SchedulerSupport.NONE)
+public final class MulticastProcessor<T> extends FlowableProcessor<T> {
+
+    final AtomicInteger wip;
+
+    final AtomicReference<Subscription> upstream;
+
+    final AtomicReference<MulticastSubscription<T>[]> subscribers;
+
+    final AtomicBoolean once;
+
+    final int bufferSize;
+
+    final int limit;
+
+    final boolean refcount;
+
+    volatile SimpleQueue<T> queue;
+
+    volatile boolean done;
+    volatile Throwable error;
+
+    int consumed;
+
+    int fusionMode;
+
+    @SuppressWarnings("rawtypes")
+    static final MulticastSubscription[] EMPTY = new MulticastSubscription[0];
+
+    @SuppressWarnings("rawtypes")
+    static final MulticastSubscription[] TERMINATED = new MulticastSubscription[0];
+
+    /**
+     * Constructs a fresh instance with the default Flowable.bufferSize() prefetch
+     * amount and no refCount-behavior.
+     * @param <T> the input and output value type
+     * @return the new MulticastProcessor instance
+     */
+    public static <T> MulticastProcessor<T> create() {
+        return new MulticastProcessor<T>(bufferSize(), false);
+    }
+
+
+    /**
+     * Constructs a fresh instance with the default Flowable.bufferSize() prefetch
+     * amount and no refCount-behavior.
+     * @param <T> the input and output value type
+     * @param refCount if true and if all Subscribers have canceled, the upstream
+     * is cancelled
+     * @return the new MulticastProcessor instance
+     */
+    public static <T> MulticastProcessor<T> create(boolean refCount) {
+        return new MulticastProcessor<T>(bufferSize(), refCount);
+    }
+
+    /**
+     * Constructs a fresh instance with the given prefetch amount and no refCount behavior.
+     * @param bufferSize the prefetch amount
+     * @param <T> the input and output value type
+     * @return the new MulticastProcessor instance
+     */
+    public static <T> MulticastProcessor<T> create(int bufferSize) {
+        return new MulticastProcessor<T>(bufferSize, false);
+    }
+
+    /**
+     * Constructs a fresh instance with the given prefetch amount and the optional
+     * refCount-behavior.
+     * @param bufferSize the prefetch amount
+     * @param refCount if true and if all Subscribers have canceled, the upstream
+     * is cancelled
+     * @param <T> the input and output value type
+     * @return the new MulticastProcessor instance
+     */
+    public static <T> MulticastProcessor<T> create(int bufferSize, boolean refCount) {
+        return new MulticastProcessor<T>(bufferSize, refCount);
+    }
+
+    /**
+     * Constructs a fresh instance with the given prefetch amount and the optional
```

Comment:
There's no real need of having the documentation here, is there? It's not a public method and the create method already cover everything